controllers/configurationpolicy_controller.go

```diff
@@ -1873,13 +1874,15 @@ func handleKeys(
 		mapMtx.Unlock()
 
 		if keyUpdateNeeded {
-			updateNeeded = true
-
-			if strings.EqualFold(string(remediation), string(policyv1.Inform)) || isStatus {
+			if strings.EqualFold(string(remediation), string(policyv1.Inform)) {
 				return true, "", false
+			} else if isStatus {
+				statusUpdated = true
+				log.Info("Ignoring an update to the object status", "key", key)
```

Comment:
How will a user that doesn't have access to the logs know why their policy is noncompliant `statusUpdated` is now returned to determine if a spec violation should be thrown?

Just to be sure, the status doesn't actually get updated on the object because we don't use the [status subresource](https://book-v1.book.kubebuilder.io/basics/status_subresource.html), right? If that's the case, then could you please add a comment explaining that since it's not intuitive here since `existingObj.UnstructuredContent()[key] = mergedObj` happens earlier in the loop, implying the status from the policy was applied on `existingObj` and will get updated.

Reply:
AFAIK, it's not just that we don't update the status because we don't use the status updater. I think it's a deliberate choice to not have this controller modify the status of an object, because that field should almost always only be modified by the actual controller for that resource.

But I am also curious about what the compliance event looks like now.

Reply:
> AFAIK, it's not just that we don't update the status because we don't use the status updater. I think it's a deliberate choice to not have this controller modify the status of an object, because that field should almost always only be modified by the actual controller for that resource.

I agree, I just wanted to verify that.
> 
> But I am also curious about what the compliance event looks like now.
This seems very important for the user to debug this.